system/jlib/jmetrics.hpp

```diff
@@ -291,6 +289,24 @@ class jlib_decl HistogramMetric : public MetricBase
 };
 
 
+class jlib_decl ScaledHistogramMetric : public HistogramMetric
+{
+public:
+    ScaledHistogramMetric(const char *name, const char *desc, StatisticMeasure units, const std::vector<BucketDef> &bucketDefs, double _limitsToMeasurementUnitsScaleFactor, const MetricMetaData &_metaData = MetricMetaData());
+
+    virtual __uint64 queryValue() const override
+    {
+        CriticalBlock block(cs);
+        {
+            return (__uint64)((double)sum / limitsToMeasurementUnitsScaleFactor);
```

Comment:
trivial efficiency:  It doesn't really matter here because not called very often, but more efficient to multiply than divide.  (Could still have the same parameter, but save the inverse in the class).

Reply:
Change the saved scale factor to outputScaleFactor and saved the inverse of the input. Converted to multiplication.